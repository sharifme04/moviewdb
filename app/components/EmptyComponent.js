import React from 'react';
import {TouchableOpacity, View, Text} from 'react-native';
import styles from './Style';
import PropTypes from 'prop-types';

const EmptyComponent = ({navigation, title}) => {
  return (
    <View style={styles.emptyTitle}>
      <Text style={styles.title}>{title}</Text>
      <TouchableOpacity
        onPress={() => navigation.navigate('Home')}
        style={styles.backHome}>
        <Text style={styles.backHome}>Back To home</Text>
      </TouchableOpacity>
    </View>
  );
};

EmptyComponent.propTypes = {
  title: PropTypes.string,
  navigation: PropTypes.object,
};
export default EmptyComponent;
