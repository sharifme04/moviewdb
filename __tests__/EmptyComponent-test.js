import 'react-native';
import React from 'react';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import EmptyComponent from '../app/components/EmptyComponent';
import renderer from 'react-test-renderer';

const mockStore = configureMockStore([thunk]);
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
const store = mockStore({});

test('EmptyComponent', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <EmptyComponent />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
