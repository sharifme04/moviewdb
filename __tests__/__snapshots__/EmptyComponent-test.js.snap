// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyComponent 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "color": "black",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#f77a55",
        "borderRadius": 25,
        "color": "#ffffff",
        "fontSize": 20,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#f77a55",
          "borderRadius": 25,
          "color": "#ffffff",
          "fontSize": 20,
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      Back To home
    </Text>
  </View>
</View>
`;
